docs/admin/monitoring.rst

```diff
@@ -244,33 +244,44 @@ information about the shards located on the node with the following attributes:
-|                      | HTTP                                                    |
-+----------------------+---------------------------------------------------------+
-| ``HttpTotal``        | The number of total connections established via HTTP    |
-|                      | over the life time of a node                            |
-+----------------------+---------------------------------------------------------+
-| ``PsqlOpen``         | The number of currently established connections via the |
-|                      | PostgreSQL protocol                                     |
-+----------------------+---------------------------------------------------------+
-| ``PsqlTotal``        | The number of total connections established via the     |
-|                      | PostgreSQL protocol over the life time of a node        |
-+----------------------+---------------------------------------------------------+
-| ``TransportOpen``    | The number of currently established connections via the |
-|                      | transport protocol                                      |
-+----------------------+---------------------------------------------------------+
++---------------------+-------------------------------------------------+
+| Name                | Description                                     |
++=====================+=================================================+
+| ``HttpStats``       | Network statistics for HTTP protocol.           |
++---------------------+-------------------------------------------------+
+| ``PsqlStats``       | Network statistics for the PostgreSQL protocol. |
++---------------------+-------------------------------------------------+
+| ``TransportStats``  | Network statistics for the transport protocol.  |
++---------------------+-------------------------------------------------+
+
+``HttpStats``, ``PsqlStats`` and ``TransportStats`` return a `CompositeData`_
+object containing detailed network statistics for the given protocol:
+
++-----------------------+----------------------------------------------------+
+| Name                  | Description                                        |
++=======================+====================================================+
+| ``OpenConnections``   | The number of currently established connections.   |
++-----------------------+----------------------------------------------------+
+| ``TotalConnections``  | The number of total established connections.       |
++-----------------------+----------------------------------------------------+
+| ``ReceivedCount``     | The number of total messages received.             |
++-----------------------+----------------------------------------------------+
+| ``ReceivedSize``      | The number of total bytes received.                |
++-----------------------+----------------------------------------------------+
+| ``TransmittedCount``  | The number of total messages sent.                 |
```

Comment:
- Maybe `SentCount` and `SentSize`?
- `Count` -> `Messages`
- `Size` -> `Bytes`

Reply:
Rearranging mbeans and previous attributes could cause problems with users running the metrics. This will break monitoring.

Reply:
> It's nicer if the previous values come out as before if possible. This will break existing monitoring setups.

I double on this, even that I like the composite structure more. These metrics are mostly used by monitoring tools which do not really care about the structure but will just break and must be adjusted. I tend to not adding breaking changes here without strong arguments.

Reply:
How about keeping the existing methods (attributes) as they were, and add those composite structures as is (of course with the suggested renames)?
This means that we duplicate those existing open/total connection values?

Just a suggestion, I don't mind to expose all attributes in a flattened top level way.

Reply:
Let's avoid duplication, I would consider this only when deprecating the others. But I don't see a real user value in the new structure which would justify this. 
I'd suggest to flatten them out instead to avoid any break. Maybe I'm missing a good argument for the new structures.